### PR TITLE
finish question 11

### PR DESCRIPTION
**Pull request by MattyB2112**
I think the "/api/articles/?topic=" test would evaluate to the same thing as my previous "api/articles" test but I added it just in case

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -391,3 +401,40 @@ describe("GET /api/users", () => {
...
   });
 });
+describe("GET /api/articles by query", () => {
+  test("returns all articles with specified topic", () => {
+    return request(app)
+      .get("/api/articles/?topic=cats")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+        console.log(articles);
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:411]: `console.log` begone 🗑️

File: __tests__/app.test.js [hunk 2]
```diff
@@ -391,3 +401,40 @@ describe("GET /api/users", () => {
...
+        expect(articles.length).toBe(1);
+        expect(typeof articles[0].author).toBe("string");
+        expect(typeof articles[0].title).toBe("string");
+        expect(typeof articles[0].article_id).toBe("number");
+        expect(typeof articles[0].topic).toBe("string");
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:416]: You can be a bit more specific about the topic here

File: __tests__/app.test.js [hunk 3]
```diff
@@ -391,3 +401,40 @@ describe("GET /api/users", () => {
...
+      .get("/api/articles/?topic=")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+        expect(articles.length).toBe(13);
+      });
+  });
+  test("non-existent query returns status 404 and error message", () => {
+    return request(app)
+      .get("/api/articles/?topic=youguessedit-bananas")
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:434]: 🍌

File: __tests__/app.test.js [hunk 4]
```diff
@@ -391,3 +401,40 @@ describe("GET /api/users", () => {
...
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe("not found");
+      });
+  });
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:439]: Can you add another 200 test where you check if you look for articles with a valid topic, but it has no articles, you get an empty array back 🔍

File: models/topics.model.js [hunk 5]
```diff
@@ -18,14 +18,27 @@ exports.fetchArticleById = (id) => {
...
+        return rows;
+      });
+  } else
+    return db
+      .query(
+        `SELECT articles.author, articles.title,articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY articles.created_at ASC`
+      )
+      .then(({ rows }) => {
+        return rows;
+      });
```
Comment (nc-katherine-hurst) [inline models/topics.model.js:41]: This approach works for the moment, but wouldn't if you were searching for articles with an existing topic but with no articles (which you'd expect an empty array for) 
You will need to check if the topic exists for this to work - make sure this is done using a query to the database 📝 
Can you please add this functionality and resubmit the ticket 🎫

**Review by nc-katherine-hurst — COMMENTED** [review]
Hi Matt, a strong start but a bit more functionality is needed! Could you please have a look at my feedback and resubmit this PR
